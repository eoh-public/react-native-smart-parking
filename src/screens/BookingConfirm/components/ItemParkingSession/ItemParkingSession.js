import React, { memo } from 'react';
import { View } from 'react-native';

import Text from '../../../../commons/Text';
import { Colors } from '../../../../configs';
import { styles } from './styles';

const ItemParkingSession = memo(({ title, value, isPrimary }) => {
  return (
    <View style={styles.container}>
      <Text type="Body" size={14} color={Colors.Gray8}>
        {title}
      </Text>
      <Text
        size={14}
        type="Body"
        color={isPrimary ? Colors.Primary : Colors.Gray9}
      >
        {value}
      </Text>
    </View>
  );
});

export default ItemParkingSession;
