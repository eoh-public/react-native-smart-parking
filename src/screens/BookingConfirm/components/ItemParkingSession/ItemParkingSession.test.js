import React from 'react';
import renderer, { act } from 'react-test-renderer';
import ItemParkingSession from './ItemParkingSession';

describe('Test ItemParkingSession', () => {
  const parking_hours = 'parking_hours';

  let tree;

  test('render item parking session', async () => {
    act(() => {
      tree = renderer.create(
        <ItemParkingSession
          title={'total_parking_hours'}
          value={parking_hours}
        />
      );
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('render item parking session with isPrimary', async () => {
    act(() => {
      tree = renderer.create(
        <ItemParkingSession
          title={'total_parking_hours'}
          value={parking_hours}
          isPrimary
        />
      );
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
