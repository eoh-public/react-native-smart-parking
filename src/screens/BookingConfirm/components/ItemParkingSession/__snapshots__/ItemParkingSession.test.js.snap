// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ItemParkingSession render item parking session 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 16,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 14,
          "lineHeight": 22,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {},
      ]
    }
  >
    total_parking_hours
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#262626",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 14,
          "lineHeight": 22,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {},
      ]
    }
  >
    parking_hours
  </Text>
</View>
`;

exports[`Test ItemParkingSession render item parking session with isPrimary 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 16,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 14,
          "lineHeight": 22,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {},
      ]
    }
  >
    total_parking_hours
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#00979D",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 14,
          "lineHeight": 22,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {},
      ]
    }
  >
    parking_hours
  </Text>
</View>
`;
